From 8a1025ca2e9e6a20086f6edb1c301b2bdffc7363 Mon Sep 17 00:00:00 2001
From: "Neal H. Walfield" <neal@sequoia-pgp.org>
Date: Thu, 18 Jul 2024 23:04:31 +0200
Subject: [PATCH 3/3] Fix the autocrypt test's time.

  - When running the autocrypt test, don't use the current time, but
    set it to a fixed time.  This prevents failures due to an expired
    certificate.
---
 tests/sq-autocrypt.rs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/sq-autocrypt.rs b/tests/sq-autocrypt.rs
index c50f715..b19a81f 100644
--- a/tests/sq-autocrypt.rs
+++ b/tests/sq-autocrypt.rs
@@ -17,7 +17,9 @@ pub fn time_as_string(t: DateTime<Utc>) -> String {
 #[test]
 fn sq_autocrypt_import() -> Result<()>
 {
-    let sq = Sq::new();
+    let t = chrono::DateTime::parse_from_str("20240304T0100z", "%Y%m%dT%H%M%#z")
+        .expect("valid date");
+    let sq = Sq::at(t.into());
 
     let manifest_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
     let eml = manifest_dir.join("tests").join("data").join("autocrypt")
-- 
2.46.0

